### ISSUE TITLE
vcpkg-init.sh fail in /bin/sh

### ISSUE THREAD

**Issue by iskiselev** [microsoft/vcpkg#41163]
**Describe the bug**
vcpkg-init.sh fail in `/bin/sh` with: `/bin/sh: 85: /tmp/vcpkg-init.sh: Syntax error: Bad function name`. It started recently, most probably after rename function to vcpkg-shell. Is - restricted for function name in sh shell?
If script is not intended to be run with just sh, should first line be updated?

**Environment**
- OS: Ubuntu 22

**To Reproduce**
Run next script:
```
curl -L https://aka.ms/vcpkg-init.sh --output /tmp/vcpkg-init.sh \
&& chmod a+x /tmp/vcpkg-init.sh \
&& /bin/sh -c ". /tmp/vcpkg-init.sh"
```

**Expected behavior**
Script runs without issues

**Failure logs**
Script fails with `/bin/sh: 85: /tmp/vcpkg-init.sh: Syntax error: Bad function name`

**Comment by iskiselev** [comment]
@BillyONeal, can we rename vcpkg-shell to vcpkgshell or something else that would be POSIX0compliant and work with /bin/sh?

**Comment by BillyONeal** [comment]
I think "change the name of the shell function we expose" and "make vcpkg-init work" are different asks here. I agree that we should make the one-liner installer work with /bin/sh, I'm less convinced that we should make our shell function work there because I've never seen a user set that as their interactive shell, and the shell-manipulating features kind of assume an interactive shell.

Are you trying to actually use the shell interactive parts here somehow or do you just want the one liner installer to work?

Thanks!

**Comment by iskiselev** [comment]
@BillyONeal, vcpkg-init.sh worked for instaling before function was renamed in sh shell. My docker file called it to install vcpkg into docker image. It has been broken when `vcpkg-shell` function was introduced in that script.
I'm interested in restoring vpkg-init serving ans vcpkg install in `sh`, which servers as default shell in many docker images. I don't need any interactive features to be supported there.

**Comment by BillyONeal** [comment]
@iskiselev Let me put this another way: Do you expect commands that mutate the shell's environment like `activate` to work?

**Comment by iskiselev** [comment]
@BillyONeal, no, I don't expect activate to work, but I expect calling `. vcpkg-init.sh` to install vcpkg binaries in some default location.

**Comment by BillyONeal** [comment]
As a workaround for now the classic git clone plus bootstrap plan should work 👍

**Comment by iskiselev** [comment]
Currently I use even easier workaround: `/bin/bash -c ". /tmp/vcpkg-init.sh" ` - luckily, bash presents in parent image, thoug is not used as default shell.

**Comment by BillyONeal** [comment]
not stale
